catalyst-gateway/bin/src/service/api/cardano/cip36/endpoint.rs

```diff
@@ -1,29 +1,79 @@
 //! Implementation of the GET `/cardano/cip36` endpoint
 
-use std::time::Duration;
-
-use poem::http::HeaderMap;
-use tokio::time::sleep;
+use poem::http::{HeaderMap, StatusCode};
+use tracing::error;
 
+use self::cardano::{hash28::HexEncodedHash28, query::stake_or_voter::StakeAddressOrPublicKey};
 use super::{
     cardano::{self},
-    response, NoneOrRBAC, SlotNo,
+    filter::{get_registration_given_stake_key_hash, get_registration_given_vote_key, snapshot},
+    response, SlotNo,
+};
+use crate::{
+    db::index::session::CassandraSession,
+    service::{
+        api::cardano::cip36::response::AllRegistration,
+        common::{self},
+    },
 };
-use crate::service::common::{self};
 
 /// Process the endpoint operation
 pub(crate) async fn cip36_registrations(
-    _lookup: Option<cardano::query::stake_or_voter::StakeOrVoter>, _asat: Option<SlotNo>,
+    lookup: Option<cardano::query::stake_or_voter::StakeOrVoter>, asat: Option<SlotNo>,
     _page: common::types::generic::query::pagination::Page,
-    _limit: common::types::generic::query::pagination::Limit, _auth: NoneOrRBAC,
-    _headers: &HeaderMap,
+    _limit: common::types::generic::query::pagination::Limit, _headers: &HeaderMap,
 ) -> response::AllRegistration {
-    // Dummy sleep, remove it
-    sleep(Duration::from_millis(1)).await;
+    let Some(session) = CassandraSession::get(true) else {
+        error!("Failed to acquire db session");
+        return AllRegistration::unprocessable_content(vec![poem::Error::from_string(
```

Comment:
I think it should be a `ServiceUnavailable` response.
`UnprocessableContent` is used for invalid request arguments according.